2-Intelligent-Agents/c_cleaningRobot.js

```diff
@@ -233,7 +234,144 @@ function makeTableControlledDiagram() {
+                                else{ score_agent2 = score_agent2-10; }
+                                if(diagram_agent2.world.floors[0].dirty || diagram_agent2.world.floors[1].dirty)
+                                  { score_agent2 = score_agent2-5-parseInt(parameters[0]); }
+                                // extra score reduction due to agent2 working for a shorter duration
+                                performance_agent2.push(score_agent2);
+                                diagram_agent2.world.simulate(action);
+                                renderWorld(diagram_agent2);
+                                renderAgentPercept(diagram_agent2, percept);
+                                renderAgentAction(diagram_agent2, action);},table[1]*1000);
+        }
+
+    //print performace scores
+    //setInterval(plotPerformance(performance_agent1, performance_agent2),STEP_TIME_MS);
+    setInterval(function(){ plotPerformance(performance_agent1, performance_agent2); },STEP_TIME_MS);
+
+    //get reader defined parameters
+    function getRulesFromPage() {
+        let table = d3.select("#performance-controlled-diagram table");
+        let dirt_freq = table.select("[data-action=dirt-freq] select").node().value;
+        let speed_agent1 = table.select("[data-action=speed-agent1] select").node().value;
+        let speed_agent2 = table.select("[data-action=speed-agent2] select").node().value;
+        let interval_agent2 = table.select("[data-action=interval-agent2] select").node().value;
+        return [dirt_freq, speed_agent1, speed_agent2, interval_agent2]
+    }
+
+    function makefloordirty() {
+        floorNumber = Math.floor(Math.random() * 2);
+        diagram_agent1.world.markFloorDirty(floorNumber);
+        diagram_agent1.floors[floorNumber].attr('class', 'dirty floor');
+        diagram_agent2.world.markFloorDirty(floorNumber);
+        diagram_agent2.floors[floorNumber].attr('class', 'dirty floor');
+    }
+    setInterval(makefloordirty,parameters[0]*1000);
+    
+    //control agent1's state
+    update_agent1();
+    setInterval(update_agent1, STEP_TIME_MS);
+
+    //control agent2's state
+    function controlAgent2(){     
+        clearInterval(agent2_interval);
+        setTimeout(function(){ agent2_interval = setInterval(update_agent2,1000); },5000);
+            
+    }
+    controlAgent2();
+    setInterval(controlAgent2,5000+parameters[3]*1000) //parameter[3] defines the working interval of agent2
+
+}   
+
+//Plotting the performance of the agents on a dynamic line chart
+var label = [0,1,2,3]; 
+var index = 3;
+function plotPerformance(performanceAgent1, performanceAgent2){
+    var performance_agent1 = performanceAgent1;
+    var performance_agent2 = performanceAgent2;
+    var ctx = document.getElementById('chartContainer').getContext('2d');
+    if(performance_agent1.length != 0  || performance_agent2.length != 0){
+        var chart = new Chart(ctx, {
```

Comment:
Is there no way to reuse the Chart object? The animations are *very* distracting!

Reply:
I tried to work on it as it was previously mentioned by you, but was unable to do so. I don't know if it is possible to do so.